src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Identity.Client.Extensibility;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Linq;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.IdentityModel.Abstractions;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    /// <summary>
+    /// Class to initialize a managed identity and identify the service.
+    /// Original source of code: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs
+    /// </summary>
+    internal class ManagedIdentityClient
+    {
+        internal const string MsiUnavailableError =
+            "Authentication with managed identity is unavailable. No managed identity endpoint found.";
```

Comment:
We need a wiki for this. There could be several reasons for this failure. 

- MSI not available in a azure resource
- MSI not enabled in a azure resource 
- MSI enabled but not a MSL supported resource (new MSI Source?)

A wiki mentioning these and how to turn in MSI would help

Reply:
The wiki I added at https://aka.ms/msal-managed-identity will list the common exceptions or the failure reasons. This error message will be removed with the integration of IMDS.

Reply:
The link format is usually: https://aka.ms/msal-net-managed-identity